docs/rules/no-extraneous-dependencies.md

```diff
@@ -31,7 +37,13 @@ If you are using JavaScript configuration (e.g., `.eslintrc.js`), then you can u
 "import/no-extraneous-dependencies": ["error", {"devDependencies": /test|spec/}]
 ```
 
-When using a regular expression, the setting will be activated if the name of the file being linted matches the given regular expression. For example, the above configurations will allow the import of `devDependencies` in files whose names include `test` or `spec`.
+When using a glob or regular expression, the setting will be activated if the name of the file being linted matches the given glob or regular expression. In addition, you can use an array to specify multiple globs or regular expressions:
+
+```js
+"import/no-extraneous-dependencies": ["error", {"devDependencies": [/test/, '/spec/', '*.test.js', '*.spec.js']}]
+```
+
+When using an array of globs or regular expressions, the setting will be activated if the name of the file being linted matches a single glob or regular expression in the array.
```

Comment:
> ['*.test.js', '*.spec.js'] I'd want that to be additive. In the case of ['*.test.js', '!**/fixtures/**'], I'd want that to be subtractive.

That is the case. Imagine

```js
[
  '**/*.test.js', // All the file ending with '.test.js'
  'test/**/*.js', // + All the files in test/
  '!**/fixtures/**' // Take all the files selected by the previous two selectors,
                    // and remove those in a fixtures (sub-)folder
]
```

> This is approaching a point of complexity that may not be best for this plugin to handle.

I'm not sure I agree, but that's a subjective point a view ;)

> Maybe allow a function for configuration

The function is doable, but then we're pushing the responsibility of doing the glob/regex logic to the user. I'd use functions as a last last resort.